rag/core/ingest_handler.py

```diff
@@ -199,15 +228,37 @@ def ingest_file(self, file_data: bytes, metadata: Dict[str, Any]) -> Dict[str, A
         file_size = len(file_data)
         logger.info(f"Ingesting file: {filename}")
 
-        # Log file processing start
-        self.process_logger.log_file_processing(
-            filename,
-            "started",
-            {
-                "size_bytes": file_size,
-                "content_type": metadata.get("content_type", "unknown"),
-            },
-        )
+        # Initialize event logger if observability is available
+        event_logger = None
+        if EventLogger and hash_config and save_config_snapshot:
+            try:
```

Comment:
I'll make the same claim here that instantiation of th EventLogger should be a requirement. Any failures to do that should be propagated to the user. As such, remove the try : except block and get rid of all the code that checks for the existence of the EventLogger instance in subsequent code.